test/e2e/windows/eviction.go

```diff
@@ -0,0 +1,170 @@
+
+		pod1 := &v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "pod1",
+			},
+			Spec: v1.PodSpec{
+				Containers: []v1.Container{
+					{
+						Name:  "pod1",
+						Image: imageutils.GetE2EImage(imageutils.ResourceConsumer),
+						Resources: v1.ResourceRequirements{
+							Requests: v1.ResourceList{
+								v1.ResourceMemory: *resource.NewQuantity(500*1024*1024, resource.BinarySI),
+							},
+						},
+						Command: []string{
+							"/bin/testlimit.exe",
+							"-accepteula",
+							"-d",
+							"100Mb",
+							"-e",
+							"5",
+							"20000s",
+							"-c",
+							"5"},
+					},
+				},
+				NodeSelector: map[string]string{
+					"kubernetes.io/os": "windows",
+				},
+				NodeName: node.Name,
+			},
+		}
+		_, err = f.ClientSet.CoreV1().Pods(f.Namespace.Name).Create(ctx, pod1, metav1.CreateOptions{})
+		framework.ExpectNoError(err)
+
+		ginkgo.By("Scheduling another pod with 500MI requests that will consume more memory")
+		chunks := int((nodeMem.capacity.Value() - nodeMem.hardEviction.Value()) / (100 * 1024 * 1024))
+		framework.Logf("Pod2 will consume %d chunks of 100Mi", chunks)
+		pod2 := &v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "pod2",
+			},
+			Spec: v1.PodSpec{
+				Containers: []v1.Container{
+					{
+						Name:  "pod2",
+						Image: imageutils.GetE2EImage(imageutils.ResourceConsumer),
+						Resources: v1.ResourceRequirements{
+							Requests: v1.ResourceList{
+								v1.ResourceMemory: *resource.NewQuantity(500*1024*1024, resource.BinarySI),
+							},
+						},
+						Command: []string{
+							"/bin/testlimit.exe",
+							"-accepteula",
+							"-d",
+							"100Mb",
+							"-e",
+							"5",
+							"20000s",
+							"-c",
+							strconv.Itoa(chunks)},
+					},
+				},
+				NodeSelector: map[string]string{
+					"kubernetes.io/os": "windows",
+				},
+				NodeName: node.Name,
+			},
+		}
+		_, err = f.ClientSet.CoreV1().Pods(f.Namespace.Name).Create(ctx, pod2, metav1.CreateOptions{})
+		framework.ExpectNoError(err)
+
+		ginkgo.By("Waiting for pods to start running")
+		e2epod.WaitForPodsRunningReady(ctx, f.ClientSet, f.Namespace.Name, 2, 0, 3*time.Minute)
+
+		framework.Logf("Waiting for pod2 to get evicted")
+		gomega.Eventually(ctx, func() bool {
+			eventList, err := f.ClientSet.CoreV1().Events(f.Namespace.Name).List(ctx, metav1.ListOptions{})
+			framework.ExpectNoError(err)
+			for _, e := range eventList.Items {
+				// Look for an event that shows FailedScheduling
+				if e.Type == "Warning" && e.Reason == "Evicted" && strings.Contains(e.Message, "pod2") {
+					framework.Logf("Found %+v event with message %+v", e.Reason, e.Message)
+					return true
+				}
+			}
+			return false
+		}, 5*time.Minute, 10*time.Second).Should(gomega.BeTrue())
+
+		ginkgo.By("Waiting for node.kubernetes.io/memory-pressure taint to be removed")
+		// ensure e2e test framework catches the memory-pressure taint
+		time.Sleep(20 * time.Second)
```

Comment:
Could you move this to a constant at the top of this file? And name it. If we hit any flakes it would be nice to be able to bump this easily.